### PR TITLE
Facility search to use es query

### PR DESCRIPTION
**Pull request by lavanyareddyreddy**
#### What tests were added (if relevant), and do all tests (rspec, karma, and integration CALS tests) pass?
-yes
92.42% Statements 61/66 0% Branches 0/2 86.36% Functions 19/22 92.42% Lines 61/66

#### Which User story does this relate to?
-#143905057

#### Brief feature description
- search to use new api endpoin call(post).
- search page should build elasticsearch query

#### Does this PR block/effect other stories?
- no

#### Do any commands needs to be run in order to run the project with this PR?
- npm run karma 

#### Code coverage percent diff as a result of this PR?
- n/a

#### JS errors diff as a result of this PR?
-n/a

### CODE REVIEW

**Comment by ratneshraval** [comment]
1. field names coming in are not matching with fields in the index (county vs fac_co_nbr).
2. Test coverage is too low for added code.

File: app/controllers/facilities_controller.rb [hunk 1]
```diff
@@ -2,33 +2,32 @@
 
 class FacilitiesController < CalsBaseController
   around_action :hypernova_render_support
+  skip_before_action :verify_authenticity_token, only: [:search]
```
Comment (ratneshraval) [inline app/controllers/facilities_controller.rb]: Let's figure out how we can remove this line and keep "verify_authenticity_token"

**Comment by lavanyareddyreddy** [comment]
1. Field names wont match because once elastic search returns result we use serializer.
2. updated with spec tests.

File: app/controllers/facilities_controller.rb (hunk 1 above)
Comment (lavanyareddyreddy) [inline app/controllers/facilities_controller.rb]: we need to set CSRF token in JS and then we can remove skip_before_action.

**Comment by ratneshraval** [comment]
@lavanyareddyreddy  can you explain 1? I meant request field names are not matching. Which returns inaccurate results.

**Comment by lavanyareddyreddy** [comment]
@ratneshraval : updated county and type column names to match field names (fac_co_nbr, fac_type). please check my commit 35651b0

**Comment by anuroop15** [comment]
why not ? We don't have separate US for this refactoring. So I committed those changes here. I'm writing some testcases for this US and thought of modifying the folder structure.

**Comment by lauherkcambria** [comment]
if there isn't a user story for this, shouldn't we make one? I feel that it's not a good call putting in unrelated commits to a specific feature branch

**Comment by ratneshraval** [comment]
@anuroop15 you don't need story to modify the structure. You can add chore if you need to. I recommend that you only modify structure of files you need to add test cases for this branch/story and rest of them modify when you take up the chore in a separate branch.

**Comment by lauherkcambria** [comment]
@ratneshraval wouldn't it be better to have all structure modifications @anuroop15 is doing be apart of one commit that is on a PR related to a single chore? more granularity that way

**Comment by ratneshraval** [comment]
What Keith said works too. I didn't propose that option in case it becomes double work. You can write tests with current structure and then re modify structure of all in new PR.

**Comment by anuroop15** [comment]
cool!! I will create a chore for refactoring and keep separate PR.

**Comment by ratneshraval** [comment]
Ruby code coverage dropped below 90%. Something to consider and improve.

**Comment by lauherkcambria** [comment]
@lavanyareddyreddy @ratneshraval nice!!!! before we merge this lets sit down tomorrow morning and get the coverage back to 100 👍

**Comment by ratneshraval** [comment]
Please pull dev branch in here so you don't have to recreate tests Keith already added.

**Comment by lavanyareddyreddy** [comment]
@ratneshraval @lauherkcambria - merged Development branch, fixed failing tests, code coverage shows 100%.

**Comment by lauherkcambria** [comment]
noice 👍 🥝

**Pull request merged**